nixos/modules/programs/obs-studio.nix

```diff
@@ -0,0 +1,37 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
+
+let cfg = config.programs.obs-studio;
+in {
+  options.programs.obs-studio = {
+    enable = mkEnableOption "OBS Studio program";
+
+    package = mkPackageOption pkgs "obs-studio" { example = "obs-studio"; };
+
+    enableVirtualCamera = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Installs and sets up the v4l2loopback kernel module, necessary for OBS
+        to start a virtual camera.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    boot = mkIf cfg.enableVirtualCamera {
```

Comment:
I think `kernelModules = [ "v4l2loopback" ];` is also necessary? At least from my testing.

Reply:
you're right! gonna add it